controllers/feeds/detailFeed.js

```diff
@@ -27,6 +28,36 @@ module.exports = async (req, res) => {
   const myAnonymousUser = await UsersFunction.findAnonymousUserId(User, req.userId, {raw: true});
 
   const newItem = {...feedItem};
+  // add karma score
+  let actors = [newItem.actor.id];
+  let comments = newItem.latest_reactions?.comment;
+  const collectUserIds = (comment) => {
+    if (comment?.user?.id && !actors.includes(comment.user.id)) {
+      actors.push(comment.user.id);
+    }
+    comment?.latest_children?.comment?.forEach(collectUserIds);
+  };
+  comments.forEach(collectUserIds);
+
+  const karmaScores = await UsersFunction.getUsersKarmaScore(User, actors);
+  // add karma score in feed detail
+  const user = karmaScores.find((user) => user.user_id === newItem.actor.id);
+  newItem.karma_score = roundingKarmaScore(user?.karma_score || 0);
+  // add karma score in comment
+  comments = comments.map((comment) => {
+    const user = karmaScores.find((user) => user.user_id === comment.user_id);
+    comment.karma_score = roundingKarmaScore(user?.karma_score || 0);
+    let sub_comments = comment.latest_children?.comment?.map((sub_comment) => {
+      const user = karmaScores.find((user) => user.user_id === sub_comment.user_id);
+      sub_comment.karma_score = roundingKarmaScore(user?.karma_score || 0);
+      return sub_comment;
+    });
+    comment.latest_children.comment = sub_comments;
+    return comment;
+  });
+
+  newItem.latest_reactions.comment = comments;
+
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for collecting user IDs and calculating karma scores seems to be correct. However, the code could be optimized by reducing the number of times we search through `karmaScores` array using `find`. This operation has a time complexity of O(n), which can be costly if the number of users is large. Instead, consider creating a map with user IDs as keys and their respective karma scores as values. This way, you can access the karma score of any user in constant time.

Here's how you can do it:

```diff
-  const karmaScores = await UsersFunction.getUsersKarmaScore(User, actors);
+  const karmaScoresArray = await UsersFunction.getUsersKarmaScore(User, actors);
+  const karmaScores = new Map(karmaScoresArray.map(user => [user.user_id, user.karma_score]));

-  const user = karmaScores.find((user) => user.user_id === newItem.actor.id);
+  const userKarmaScore = karmaScores.get(newItem.actor.id);

-  const user = karmaScores.find((user) => user.user_id === comment.user_id);
+  const userKarmaScore = karmaScores.get(comment.user_id);

-  const user = karmaScores.find((user) => user.user_id === sub_comment.user_id);
+  const userKarmaScore = karmaScores.get(sub_comment.user_id);
```

Also, please ensure that error handling is properly implemented in the `getUsersKarmaScore` function. If there's an issue fetching the karma scores, it should not break the entire feed detail functionality.


<!-- This is an auto-generated comment by OSS CodeRabbit -->